libexec/cli/list.exec

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+# 
+# Copyright (c) 2017, SingularityWare, LLC. All rights reserved.
+#
+# This software is licensed under a 3-clause BSD license.  Please
+# consult LICENSE file distributed with the sources of this project regarding
+# your rights to use or distribute this software. 
+#
+# 
+
+
+## Basic sanity
+if [ -z "$SINGULARITY_libexecdir" ]; then
+    echo "Could not identify the Singularity libexecdir."
+    exit 1
+fi
+
+## Load functions
+if [ -f "$SINGULARITY_libexecdir/singularity/functions" ]; then
+    . "$SINGULARITY_libexecdir/singularity/functions"
+else
+    echo "Error loading functions: $SINGULARITY_libexecdir/singularity/functions"
+    exit 1
+fi
+
+## Init Singularity environment
+if [ -f "$SINGULARITY_sysconfdir/singularity/init" ]; then
+    . "$SINGULARITY_sysconfdir/singularity/init"
+fi
+
+
+if ! USERID=`id --real --user`; then
+    message ERROR "Could not ascertain user ID\n"
+    exit 255
+fi
+
+
+while true; do
+    case ${1:-} in
+        -h|--help|help)
+            if [ -e "$SINGULARITY_libexecdir/singularity/cli/$SINGULARITY_COMMAND.help" ]; then
+                cat "$SINGULARITY_libexecdir/singularity/cli/$SINGULARITY_COMMAND.help"
+            else
+                message ERROR "No help exists for this command\n"
+                exit 1
+            fi
+            exit
+        ;;
+        -a|--all)
+            if [ "$USERID" = 0 ]; then
+                USERID="*"
+                shift
+            else
+                message ERROR "Must be root to list all\n"
+                exit 1
+            fi
+        ;;
+        -u|--user)
+            if [ "$USERID" = 0 ]; then
+                shift
+                USERID=${1:-}
+                shift
+            else
+                message ERROR "Must be root to list with -u/--user option\n"
+                exit 1
+            fi
+        ;;
+        -*)
+            message ERROR "Unknown option: ${1:-}\n"
+            exit 1
+        ;;
+        *)
+            break;
+        ;;
+    esac
+done
+
+
+# This is a configurable option, needs to be set accordingly!
+SESSION_GLOB="/tmp/.singularity-daemon-${USERID}/*"
+
```

Comment:
I think the loose coupling between the process and the PID file is a pretty serious problem that needs to be resolved.  See my comment below.